#!/usr/bin/env python
PACKAGE = "lane_follow_igvc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('enable',        bool_t,   0, 'Enable',        False)
gen.add('gain',        double_t,   0, 'Gain',             1.7,   0,   10)
gen.add('speed',       double_t,   0, 'Speed',            0.5,   0,    5)
gen.add('frame_skip',     int_t,   0, 'Frame Skip',         4,   1,   10)

# Add hue, sat, and val low and high
gen.add('hue_low', int_t, 0, 'Hue Low', 56, 0, 179)
gen.add('hue_high', int_t, 0, 'Hue High', 179, 0, 179)
gen.add('sat_low', int_t, 0, 'Sat Low', 109, 0, 255)
gen.add('sat_high', int_t, 0, 'Sat High', 169, 0, 255)
gen.add('val_low', int_t, 0, 'Value Low', 105, 0, 255)
gen.add('val_high', int_t, 0, 'Value High', 177, 0, 255)

exit(gen.generate(PACKAGE, "lane_follow_igvc", "LineFollowDynCfg"))

